optuna/trial.py

```diff
@@ -104,17 +106,30 @@ class Trial(BaseTrial):
             A :class:`~optuna.study.Study` object.
         trial_id:
             A trial ID that is automatically generated.
-
+        predefined_search_space:
+            A search space defined by a sampler before executing the target object function.
+            The final search space of this trial is the mixture of this space and
+            the one that defined by execution of the target object function.
+        predefined_params:
+            A suggested parameters by a sampler before executing the target object function.
     """
 
-    def __init__(self, study, trial_id):
-        # type: (Study, int) -> None
+    def __init__(
+            self,
+            study,  # type: Study
+            trial_id,  # type: int
+            predefined_search_space=None,  # type: Optional[Dict[str, BaseDistribution]]
```

Comment:
I'm sorry if I'm misunderstanding.
I think this variable isn't needed in Trial object. It's enough if `predefined_params` is given.

Reply:
In ordinal cases, as you say, this argument is unnecessary because the distributions of parameters are given via `trial.suggest_xxx()` method calls in object function.
However, in theory, `BaseSampler.define_relative_search_space()` can define arbitrary search space and distributions in the search space can be inconsistent with ones that would be specified by users.
`Trial` uses `predefined_search_space` for detecting the inconsistency.

Reply:
Thanks! I understand. It's good to detect the inconsistencies.